<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>741</width>
    <height>675</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>741</width>
    <height>675</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>741</width>
    <height>675</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="styleSheet">
   <string notr="true">background-color:  #F4F4EF;
</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QTabWidget" name="tabWidget">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>30</y>
      <width>701</width>
      <height>581</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <bold>true</bold>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QTabWidget::pane
{
	border-radius: 5px;
	border-width: 2px;
	border-style: solid;
	border-color: #20232A;
	background: #F4F4EF;
}

QTabBar::tab
{
	border-radius: 5px;	
	background: #F4F4EF;
	border-width: 2px;
	border-style: solid;
	border-color: #20232A;
 	height: 30px;
	width: 346px;
	color: #20232A;
}

QTabBar::tab:selected
{
	background:#20232A;
	color: #F4F4EF;
}

QTabBar::tab:hover
{
	background:#ACBEBE;
	color: #F4F4EF;
}

QTabBar::tab:pressed {
background-color: qlineargradient(x1: 0, y1: 0, x2: 0, y2: 1,
                                      stop: 0 #ACBEBE, stop: 1 #20232A);
}
</string>
    </property>
    <property name="currentIndex">
     <number>1</number>
    </property>
    <widget class="QWidget" name="labyrinth">
     <attribute name="title">
      <string>MAZE</string>
     </attribute>
     <widget class="QLabel" name="logo1">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>20</y>
        <width>161</width>
        <height>151</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">image: url(:/icon/maze.png);</string>
      </property>
      <property name="text">
       <string/>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_openfile">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>190</y>
        <width>141</width>
        <height>40</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <bold>true</bold>
       </font>
      </property>
      <property name="styleSheet">
       <string notr="true">QPushButton
{
	border-radius: 5px;
	border-width: 2px;
	border-style: solid;
	border-color: #20232A;
	background-color: #20232A;
	color: #F4F4EF;
}

QPushButton:hover
{
	background:#ACBEBE;
	color: #F4F4EF;
}

QPushButton:pressed {
background-color: qlineargradient(x1: 0, y1: 0, x2: 0, y2: 1,
                                      stop: 0 #ACBEBE, stop: 1 #20232A);
}
</string>
      </property>
      <property name="text">
       <string>OPEN FILE</string>
      </property>
     </widget>
     <widget class="QLabel" name="fileName">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>240</y>
        <width>141</width>
        <height>21</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">border-radius: 5px;
border-width: 2px;
border-style: solid;
border-color: #20232A;
color: #A01D26;</string>
      </property>
      <property name="text">
       <string/>
      </property>
     </widget>
     <widget class="Line" name="line_3">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>455</y>
        <width>141</width>
        <height>20</height>
       </rect>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
     <widget class="Line" name="line_4">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>265</y>
        <width>141</width>
        <height>20</height>
       </rect>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
     <widget class="QLabel" name="label_rows">
      <property name="geometry">
       <rect>
        <x>40</x>
        <y>290</y>
        <width>70</width>
        <height>40</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">color:#20232A;</string>
      </property>
      <property name="text">
       <string>ROWS</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
      </property>
     </widget>
     <widget class="QLabel" name="label_columns">
      <property name="geometry">
       <rect>
        <x>40</x>
        <y>350</y>
        <width>70</width>
        <height>40</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">color:#20232A;</string>
      </property>
      <property name="text">
       <string>COLUMNS</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
      </property>
     </widget>
     <widget class="QSpinBox" name="spinBox_rows">
      <property name="geometry">
       <rect>
        <x>120</x>
        <y>290</y>
        <width>41</width>
        <height>40</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">QSpinBox
{
border-radius: 5px;
border-width: 2px;
border-style: solid;
border-color: #20232A;
color: #A01D26;
background-color: #F4F4EF;
}

QSpinBox::down-button  
{
  height: 0px;
  width: 0px;
}

QSpinBox::up-button  
{
  height: 0px;
  width: 0px;
}
</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
      <property name="minimum">
       <number>0</number>
      </property>
      <property name="maximum">
       <number>50</number>
      </property>
     </widget>
     <widget class="QSpinBox" name="spinBox_columns">
      <property name="geometry">
       <rect>
        <x>120</x>
        <y>350</y>
        <width>41</width>
        <height>40</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">QSpinBox
{
border-radius: 5px;
border-width: 2px;
border-style: solid;
border-color: #20232A;
color: #A01D26;
background-color: #F4F4EF;
}

QSpinBox::down-button  
{
  height: 0px;
  width: 0px;
}

QSpinBox::up-button  
{
  height: 0px;
  width: 0px;
}
</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
      <property name="minimum">
       <number>0</number>
      </property>
      <property name="maximum">
       <number>50</number>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_generate">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>410</y>
        <width>141</width>
        <height>40</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <bold>true</bold>
       </font>
      </property>
      <property name="styleSheet">
       <string notr="true">QPushButton
{
	border-radius: 5px;
	border-width: 2px;
	border-style: solid;
	border-color: #20232A;
	background-color: #20232A;
	color: #F4F4EF;
}

QPushButton:hover
{
	background:#ACBEBE;
	color: #F4F4EF;
}

QPushButton:pressed {
background-color: qlineargradient(x1: 0, y1: 0, x2: 0, y2: 1,
                                      stop: 0 #ACBEBE, stop: 1 #20232A);
}
</string>
      </property>
      <property name="text">
       <string>GENERATE</string>
      </property>
     </widget>
     <widget class="PaintWidget" name="widget_maze" native="true">
      <property name="geometry">
       <rect>
        <x>180</x>
        <y>20</y>
        <width>501</width>
        <height>501</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true"/>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_savefile">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>480</y>
        <width>141</width>
        <height>40</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <bold>true</bold>
       </font>
      </property>
      <property name="styleSheet">
       <string notr="true">QPushButton
{
	border-radius: 5px;
	border-width: 2px;
	border-style: solid;
	border-color: #20232A;
	background-color: #20232A;
	color: #F4F4EF;
}

QPushButton:hover
{
	background:#ACBEBE;
	color: #F4F4EF;
}

QPushButton:pressed {
background-color: qlineargradient(x1: 0, y1: 0, x2: 0, y2: 1,
                                      stop: 0 #ACBEBE, stop: 1 #20232A);
}
</string>
      </property>
      <property name="text">
       <string>SAVE FILE</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="cave">
     <attribute name="title">
      <string>CAVE</string>
     </attribute>
     <widget class="QLabel" name="logo">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>20</y>
        <width>161</width>
        <height>151</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">image: url(:/icon/maze.png);</string>
      </property>
      <property name="text">
       <string/>
      </property>
     </widget>
     <widget class="QLabel" name="fileName_cave">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>240</y>
        <width>141</width>
        <height>21</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">border-radius: 5px;
border-width: 2px;
border-style: solid;
border-color: #20232A;
color: #A01D26;</string>
      </property>
      <property name="text">
       <string/>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_openfile_cave">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>190</y>
        <width>141</width>
        <height>40</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <bold>true</bold>
       </font>
      </property>
      <property name="styleSheet">
       <string notr="true">QPushButton
{
	border-radius: 5px;
	border-width: 2px;
	border-style: solid;
	border-color: #20232A;
	background-color: #20232A;
	color: #F4F4EF;
}

QPushButton:hover
{
	background:#ACBEBE;
	color: #F4F4EF;
}

QPushButton:pressed {
background-color: qlineargradient(x1: 0, y1: 0, x2: 0, y2: 1,
                                      stop: 0 #ACBEBE, stop: 1 #20232A);
}
</string>
      </property>
      <property name="text">
       <string>OPEN FILE</string>
      </property>
     </widget>
     <widget class="PaintWidgetCave" name="widget_cave" native="true">
      <property name="geometry">
       <rect>
        <x>180</x>
        <y>20</y>
        <width>501</width>
        <height>501</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true"/>
      </property>
     </widget>
     <widget class="QSpinBox" name="spinBox_birth">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>380</y>
        <width>41</width>
        <height>40</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">QSpinBox
{
border-radius: 5px;
border-width: 2px;
border-style: solid;
border-color: #20232A;
color: #A01D26;
background-color: #F4F4EF;
}

QSpinBox::down-button  
{
  height: 0px;
  width: 0px;
}

QSpinBox::up-button  
{
  height: 0px;
  width: 0px;
}
</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
      <property name="minimum">
       <number>0</number>
      </property>
      <property name="maximum">
       <number>8</number>
      </property>
     </widget>
     <widget class="QLabel" name="label_birth">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>420</y>
        <width>41</width>
        <height>40</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>10</pointsize>
       </font>
      </property>
      <property name="styleSheet">
       <string notr="true">color:#20232A;</string>
      </property>
      <property name="text">
       <string>BIRTH</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QLabel" name="label_death">
      <property name="geometry">
       <rect>
        <x>70</x>
        <y>420</y>
        <width>41</width>
        <height>40</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>10</pointsize>
       </font>
      </property>
      <property name="styleSheet">
       <string notr="true">color:#20232A;</string>
      </property>
      <property name="text">
       <string>DEATH</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QSpinBox" name="spinBox_death">
      <property name="geometry">
       <rect>
        <x>70</x>
        <y>380</y>
        <width>41</width>
        <height>40</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>13</pointsize>
       </font>
      </property>
      <property name="styleSheet">
       <string notr="true">QSpinBox
{
border-radius: 5px;
border-width: 2px;
border-style: solid;
border-color: #20232A;
color: #A01D26;
background-color: #F4F4EF;
}

QSpinBox::down-button  
{
  height: 0px;
  width: 0px;
}

QSpinBox::up-button  
{
  height: 0px;
  width: 0px;
}
</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
      <property name="minimum">
       <number>0</number>
      </property>
      <property name="maximum">
       <number>8</number>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_autostep">
      <property name="geometry">
       <rect>
        <x>120</x>
        <y>480</y>
        <width>41</width>
        <height>40</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>10</pointsize>
        <bold>true</bold>
       </font>
      </property>
      <property name="styleSheet">
       <string notr="true">QPushButton
{
	border-radius: 5px;
	border-width: 2px;
	border-style: solid;
	border-color: #20232A;
	background-color: #20232A;
	color: #F4F4EF;
}

QPushButton:hover
{
	background:#ACBEBE;
	color: #F4F4EF;
}

QPushButton:pressed {
background-color: qlineargradient(x1: 0, y1: 0, x2: 0, y2: 1,
                                      stop: 0 #ACBEBE, stop: 1 #20232A);
}
</string>
      </property>
      <property name="text">
       <string>AUTO
ms</string>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_nextstep">
      <property name="geometry">
       <rect>
        <x>70</x>
        <y>480</y>
        <width>40</width>
        <height>40</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>10</pointsize>
        <bold>true</bold>
       </font>
      </property>
      <property name="styleSheet">
       <string notr="true">QPushButton
{
	border-radius: 5px;
	border-width: 2px;
	border-style: solid;
	border-color: #20232A;
	background-color: #20232A;
	color: #F4F4EF;
}

QPushButton:hover
{
	background:#ACBEBE;
	color: #F4F4EF;
}

QPushButton:pressed {
background-color: qlineargradient(x1: 0, y1: 0, x2: 0, y2: 1,
                                      stop: 0 #ACBEBE, stop: 1 #20232A);
}
</string>
      </property>
      <property name="text">
       <string>NEXT
STEP</string>
      </property>
     </widget>
     <widget class="QSpinBox" name="spinBox_rand">
      <property name="geometry">
       <rect>
        <x>120</x>
        <y>290</y>
        <width>41</width>
        <height>40</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">QSpinBox
{
border-radius: 5px;
border-width: 2px;
border-style: solid;
border-color: #20232A;
color: #A01D26;
background-color: #F4F4EF;
}

QSpinBox::down-button  
{
  height: 0px;
  width: 0px;
}

QSpinBox::up-button  
{
  height: 0px;
  width: 0px;
}
</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
      <property name="minimum">
       <number>0</number>
      </property>
      <property name="maximum">
       <number>100</number>
      </property>
     </widget>
     <widget class="QLabel" name="label_caves_cols">
      <property name="geometry">
       <rect>
        <x>69</x>
        <y>330</y>
        <width>41</width>
        <height>40</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>10</pointsize>
       </font>
      </property>
      <property name="styleSheet">
       <string notr="true">color:#20232A;</string>
      </property>
      <property name="text">
       <string>COLS</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QLabel" name="label_cavws_rows">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>330</y>
        <width>41</width>
        <height>40</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>10</pointsize>
       </font>
      </property>
      <property name="styleSheet">
       <string notr="true">color:#20232A;</string>
      </property>
      <property name="text">
       <string>ROWS</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QSpinBox" name="spinBox_caves_rows">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>290</y>
        <width>41</width>
        <height>40</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">QSpinBox
{
border-radius: 5px;
border-width: 2px;
border-style: solid;
border-color: #20232A;
color: #A01D26;
background-color: #F4F4EF;
}

QSpinBox::down-button  
{
  height: 0px;
  width: 0px;
}

QSpinBox::up-button  
{
  height: 0px;
  width: 0px;
}
</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
      <property name="minimum">
       <number>0</number>
      </property>
      <property name="maximum">
       <number>50</number>
      </property>
     </widget>
     <widget class="QSpinBox" name="spinBox_caves_cols">
      <property name="geometry">
       <rect>
        <x>70</x>
        <y>290</y>
        <width>41</width>
        <height>40</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">QSpinBox
{
border-radius: 5px;
border-width: 2px;
border-style: solid;
border-color: #20232A;
color: #A01D26;
background-color: #F4F4EF;
}

QSpinBox::down-button  
{
  height: 0px;
  width: 0px;
}

QSpinBox::up-button  
{
  height: 0px;
  width: 0px;
}
</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
      <property name="minimum">
       <number>0</number>
      </property>
      <property name="maximum">
       <number>50</number>
      </property>
     </widget>
     <widget class="QSpinBox" name="spinBox_auto">
      <property name="geometry">
       <rect>
        <x>120</x>
        <y>380</y>
        <width>41</width>
        <height>40</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>13</pointsize>
       </font>
      </property>
      <property name="styleSheet">
       <string notr="true">QSpinBox
{
border-radius: 5px;
border-width: 2px;
border-style: solid;
border-color: #20232A;
color: #A01D26;
background-color: #F4F4EF;
}

QSpinBox::down-button  
{
  height: 0px;
  width: 0px;
}

QSpinBox::up-button  
{
  height: 0px;
  width: 0px;
}
</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
      <property name="minimum">
       <number>0</number>
      </property>
      <property name="maximum">
       <number>1000</number>
      </property>
      <property name="singleStep">
       <number>50</number>
      </property>
      <property name="stepType">
       <enum>QAbstractSpinBox::DefaultStepType</enum>
      </property>
      <property name="value">
       <number>0</number>
      </property>
     </widget>
     <widget class="QLabel" name="label_rand">
      <property name="geometry">
       <rect>
        <x>120</x>
        <y>330</y>
        <width>41</width>
        <height>40</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>10</pointsize>
       </font>
      </property>
      <property name="styleSheet">
       <string notr="true">color:#20232A;</string>
      </property>
      <property name="text">
       <string>RAND</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="Line" name="line_5">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>265</y>
        <width>141</width>
        <height>20</height>
       </rect>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
     <widget class="QLabel" name="label_death_2">
      <property name="geometry">
       <rect>
        <x>110</x>
        <y>420</y>
        <width>61</width>
        <height>40</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>10</pointsize>
       </font>
      </property>
      <property name="styleSheet">
       <string notr="true">color:#20232A;</string>
      </property>
      <property name="text">
       <string>STEP ms</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_generate_cave">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>480</y>
        <width>40</width>
        <height>40</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>10</pointsize>
        <bold>true</bold>
       </font>
      </property>
      <property name="styleSheet">
       <string notr="true">QPushButton
{
	border-radius: 5px;
	border-width: 2px;
	border-style: solid;
	border-color: #20232A;
	background-color: #20232A;
	color: #F4F4EF;
}

QPushButton:hover
{
	background:#ACBEBE;
	color: #F4F4EF;
}

QPushButton:pressed {
background-color: qlineargradient(x1: 0, y1: 0, x2: 0, y2: 1,
                                      stop: 0 #ACBEBE, stop: 1 #20232A);
}
</string>
      </property>
      <property name="text">
       <string>GENE
RATE</string>
      </property>
     </widget>
     <widget class="Line" name="line_6">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>455</y>
        <width>141</width>
        <height>20</height>
       </rect>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>741</width>
     <height>24</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <customwidgets>
  <customwidget>
   <class>PaintWidget</class>
   <extends>QWidget</extends>
   <header>PaintWidget/paintwidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>PaintWidgetCave</class>
   <extends>QWidget</extends>
   <header>PaintWidget/paintwidgetcave.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
